### ISSUE TITLE
Mounting a folder doesn't appear in colima

### ISSUE THREAD

**Issue by PhoenixRe32** [abiosoft/colima#515]
### Description

I start colima and try to mount a folder from the host system so when I start a container I can mount that folder in the container
The mounted folder doesn't appear in the colima filesystem.

I have the ventura MacOS and use virtiofs and vz.

### Version

colima version HEAD-5a94ab4
git commit: 5a94ab4f098ec0fe94e6d0df8b411fb149fe26fe

runtime: docker
arch: aarch64
client: v20.10.21
server: v20.10.20
limactl version 0.14.1
qemu-img version 7.2.0

### Operating System

- [ ] macOS Intel
- [X] macOS M1
- [ ] Linux

### Reproduction Steps

1. Edit settings to add a mounted folder. In my case
```
mounts:
  - location: /var/folders
    mountPoint: /var/folders
    writable: false
```
2. Start colima

### Expected behaviour

Expect to see `/var/folders` in the colima fs but it is not there.

### Additional context

`df -h` output
```
Filesystem                Size      Used Available Use% Mounted on
devtmpfs                 10.0M         0     10.0M   0% /dev
shm                     991.9M         0    991.9M   0% /dev/shm
/dev/sda                267.2M    267.2M         0 100% /media/sda
tmpfs                   991.9M    482.6M    509.3M  49% /
tmpfs                   396.8M    396.0K    396.4M   0% /run
/dev/loop0               11.9M     11.9M         0 100% /.modloop
cgroup_root              10.0M         0     10.0M   0% /sys/fs/cgroup
/dev/disk/by-label/cidata
                          6.6M      6.6M         0 100% /mnt/lima-cidata
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /mnt/data
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /etc
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /home
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /root
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /tmp
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /usr/local
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /var/lib
mount1                  926.4G    418.8G    507.5G  45% /tmp/colima
vz-rosetta              926.4G    418.8G    507.5G  45% /mnt/lima-rosetta
mount0                  926.4G    418.8G    507.5G  45% /Users/andreasandreou
/dev/disk/by-label/data-volume
                         58.8G      1.1G     54.6G   2% /var/lib/docker
```
I expected to see an entry something like 
`:/var/folders           926.4G    443.9G    482.4G  48% /var/folders`

```
colima:~$ cd /var/folders
/bin/ash: cd: can't cd to /var/folders: No such file or directory
```

**Comment by abiosoft** [comment]
The folder you are mounting needs to either be within `$HOME` or within `/tmp/colima`, those are the only two directories mounted into the VM.

**Comment by rfay** [comment]
This is actually an issue I've been meaning to write up; this error has confounded at least one DDEV user. Is there a way that we can make colima error out when mounting something that is unavailable, instead of just providing an empty mount?

**Comment by rfay** [comment]
What happens in some cases is people are just used to being able to use a random directory for the project root. Like /some/project, instead of doing the normal thing and working inside homedir. Then DDEV sets up mounts, as you'd expect, but they're empty. In any other Docker environment they'd either work or they'd fail, but on Colima, they're just empty.

**Comment by davidolrik** [comment]
I too have this issue as I have all of my project folders on a separate case-sensitive volume.

I solved it by also adding provisioning config like so:

```yaml
provision:
  - mode: system
    script: |
      mkdir -p /Users/davidolrik
      mount /Users/davidolrik
      mkdir -p /tmp/colima
      mount /tmp/colima
      mkdir -p /Volumes/Projects
      mount /Volumes/Projects
      
mounts:
  - location: /Users/davidolrik
    writable: true
  - location: /tmp/colima
    writable: true
  - location: /Volumes/Projects
    writable: true
```

Adding just the `mounts` section updates `/etc/fstab` correctly but does not create mount points nor are they mounted automatically.
Adding the `provision` block fixes the issue. - I guess that Colima also does this for the two default volumens when `mounts` is an empty array.

**Comment by abiosoft** [comment]
@davidolrik Thanks for this, you probably just unearthed a bug. 
I just noticed the mount commands are not executed in some scenarios but I am not sure if it is the cause for this issue.

I will push a fix for it and wait for your feedbacks.

**Comment by davidolrik** [comment]
@abiosoft The fix in #521 works for me, but the mounts has now been changed to `fuse.sshfs` - is this intentional?

**Comment by davidolrik** [comment]
@abiosoft I had `mountType: sshfs` in my template, changing it to `mountType: p9` changed them back as they were in v0.5.0 - looks like you fixed an extra bug there 😊

**Comment by abiosoft** [comment]
> looks like you fixed an extra bug there 😊

Yeah, your scenario helped with reproducing it and should be fixed now. There's a new release on homebrew and you can upgrade.

If @PhoenixRe32 confirms this is fixed, we can close this and open a separate issue for what @rfay reported, that is a bit different as it is with regards to better error message when using unmounted volumes in Docker.

**Comment by davidolrik** [comment]
Already updated and removed my workaround.

**Comment by PhoenixRe32** [comment]
The Volume is mounted now

**Comment by PhoenixRe32** [comment]
Thanks

**Comment by KevinSnyderCodes** [comment]
FYI: For those that were using `/tmp/colima` instead of their home directory, this behavior was removed in 6f098aa60e0fb15bc583eb337832095b9cf32a3a and included in release [v0.9.0](https://github.com/abiosoft/colima/releases/tag/v0.9.0)

Now only the home directory is mounted by default.
